server/src/actors/project.rs

```diff
@@ -334,8 +342,13 @@ impl ProjectState {
 
     /// Returns whether this state is outdated and needs to be refetched.
     pub fn outdated(&self, config: &Config) -> bool {
+        let jitter = config.cache_timeout_jitter();
+        let factor = jitter * (self.seed * 2.0 - self.seed);
+        debug_assert!(factor >= -1.0 * self.seed && factor <= 1.0 * self.seed);
```

Comment:
Shouldn't `factor` never be negative? Otherwise you get an elapsed timedelta in the negative range.

Reply:
Also can we add an assert with less variable deps?

```
debug_assert!(factor >= 0.0 && factor <= 1.0);
```

Since both seed and jitter are capped at 1.

Reply:
Oh yeah, was refactoring this at the end and ended up with wrong math. The idea here is to apply the jitter in percent to the interval. So for a 6 minute interval with 50% jitter, the actual interval can be anywhere between 3 and 9 minutes.

Reply:
I'll need to double-check this again tomorrow. Now that I'm looking at this, `random()` returns a standard deviation, so multiplying with `jitter` this way will generate skew. I shouldn't code when jet-lagged.

Reply:
I think this is fine.

The *deviation* will change from `x` to `x * jitter`, but that's what you intended to do.

The *distribution* is still a standard distribution, just with more spread.

Generally I believe that multiplying your random value with a constant (jitter) is fine. You will create *holes* in theory, but floats already have holes. You just add more holes.